test/AquaTest.jl

```diff
@@ -0,0 +1,29 @@
+module AquaTest
+
+using Test
+using ForwardDiff
+using Aqua
+
+@testset "Aqua tests - unbound_args" begin
+    # This tests that we don't accidentally run into
+    # https://github.com/JuliaLang/julia/issues/29393
+    ua = Aqua.detect_unbound_args_recursively(ForwardDiff)
+    @test length(ua) == 6
+end
+
+@testset "Aqua tests - ambiguities" begin
+    # See: https://github.com/SciML/OrdinaryDiffEq.jl/issues/1750
+    # Test that we're not introducing method ambiguities across deps
+    ambs = Aqua.detect_ambiguities(ForwardDiff; recursive = true)
+    pkg_match(pkgname, pkdir::Nothing) = false
+    pkg_match(pkgname, pkdir::AbstractString) = occursin(pkgname, pkdir)
+    filter!(x -> pkg_match("ForwardDiff", pkgdir(last(x).module)), ambs)
+
+    @test length(ambs) == 0
```

Comment:
Seems this test fails: https://github.com/JuliaDiff/ForwardDiff.jl/actions/runs/7412737730/job/20170158384?pr=662#step:6:1288

Reply:
I wasn't able to reproduce this locally for 1.10.

One of the ambiguities that I'm seeing on 1.6, `@inline Base.convert(::Type{Dual{T,V,N}}, x) where {T,V,N} = Dual{T}(V(x), zero(Partials{N,V}))`, can be eliminated, but we would need to depend on ChainRulesCore, and define `@inline Base.convert(::Type{Dual{T,V,N}}, x::ChainRulesCore.AbstractZero) where {T,V,N} = Dual{T}(V(x), zero(Partials{N,V}))` in order to do so.

I'm not sure that adding the dependency is worth it? I suppose we could add it as an extension, but I'd rather address that in a followup PR.

Reply:
Just run the Aqua tests first, before StaticArrays etc are loaded (I think StaticArrays or some other test dependency might pull in ChainRulesCore?). If only ForwardDiff, Aqua, and Test are loaded such indirect issues should not show up.

Reply:
Done, let's see what happens. Hmm, I'm getting the same thing locally (6 ambiguities).